internal/resources/cloudprovider/resource_aws_account.go

```diff
@@ -2,36 +2,43 @@ package cloudprovider
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/grafana/terraform-provider-grafana/v3/internal/common"
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common/cloudproviderapi"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
 var (
-	resourceAWSAccountTerraformID = common.NewResourceID(common.StringIDField("stack_id"), common.StringIDField("role_arn"))
+	resourceAWSAccountTerraformID = common.NewResourceID(common.StringIDField("stack_id"), common.StringIDField("account_id"))
 )
 
 func resourceAWSAccount() *common.Resource {
 	schema := &schema.Resource{
-		CreateContext: resourceAWSAccountCreate,
-		ReadContext:   resourceAWSAccountRead,
-		UpdateContext: resourceAWSAccountUpdate,
-		DeleteContext: resourceAWSAccountDelete,
+		CreateContext: withClient[schema.CreateContextFunc](resourceAWSAccountCreate),
+		ReadContext:   withClient[schema.ReadContextFunc](resourceAWSAccountRead),
+		UpdateContext: withClient[schema.UpdateContextFunc](resourceAWSAccountUpdate),
+		DeleteContext: withClient[schema.DeleteContextFunc](resourceAWSAccountDelete),
 		Schema: map[string]*schema.Schema{
 			"stack_id": {
 				Description: "The StackID of the Grafana Cloud instance. Part of the Terraform Resource ID.",
 				Type:        schema.TypeString,
 				Required:    true,
 			},
+			"account_id": {
+				Description: "The ID computed by the Grafana Cloud Provider API for the AWS Account resource.",
```

Comment:
nit: maybe simplify it to
```suggestion
				Description: "The AWS Account ID extracted from the configured AWS IAM Role.",
```

Reply:
I renamed `account_id` to `resource_id`. The plan for now is to have our AWS Accounts API will generate its own IDs for each accounts resource, so this is distinct from the AWS Account ID.